sceptre/plan/actions.py

```diff
@@ -614,18 +614,127 @@ def diff(self, differ="difflib"):
         local_template_stream = self.stack.template.body
 
         if differ == "difflib":
-            return_val = self._diff_by_difflib(
+            response = self._diff_by_difflib(
                 remote_template_stream,
                 local_template_stream
             )
 
         elif differ == "dictdiffer":
-            return_val = self._diff_by_dictdiffer(
+            response = self._diff_by_dictdiffer(
                 remote_template_stream,
                 local_template_stream
             )
 
-        return [self.stack.external_name, return_val]
+        return [self.stack.external_name, response]
+
+    @add_stack_hooks
+    def detect_stack_drift(self):
+        """
+        Detects stack drift for a running stack.
+
+        :returns: The stack drift.
+        :rtype: List[str]
+        """
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift(detection_id)
+
+        if status == "DETECTION_COMPLETE":
+            response = self._describe_stack_resource_drifts()
+            return_value = [self.stack.external_name, response]
+        else:
+            return_value = [self.stack.external_name, status]
+
+        return return_value
+
+    def _wait_for_drift(self, detect_id):
+        """
+        Waits for drift detection to complete.
+
+        :param detect_id: The drift detection ID.
+        :type detect_id: str
+
+        :returns: The drift status.
+        :rtype: str
+        """
+        timeout = 300
+        elapsed = 0
+
+        def print_status():
```

Comment:
Not a huge fan of the nested functions. Either refactor it out or just keep it inline

Reply:
done